### PR TITLE
feat(GODT-1556): simulate reply behaviour on fakeAPI side.

### PR DESCRIPTION
**Pull request by rlejeune74**
When creating the draft we need to convert the parentID into valid reference to be stored as references/in-reply-to.
Then on message creation on the fake server we need to populate the in-reply-to/references fields.

### CODE REVIEW

File: server/backend/message.go [hunk 1]
```diff
@@ -19,6 +19,8 @@ type message struct {
 	labelIDs   []string
 	sysLabel   *string
 	attIDs     []string
+	references []string
```
Comment (jameshoulahan) [inline server/backend/message.go:22]: I don't think this needs to be stored in the server. If we have the `inReplyTo` (which you could probably call `parentID`, like `externalID`), then the references header can be created at message build time in `getHeader()` with some simple logic (checking if it's non-empty).

Comment (rlejeune74) [reply server/backend/message.go:22]: I found it more explicit to have it here to highlight what the live API is doing (or what this value reflects).
I feel like putting it in the `getHeader()`  will somehow "hide" how they are handled in the backend.

Comment (jameshoulahan) [reply server/backend/message.go:22]: But this is what he API does -- it doesn't store the references as a separate DB entry, it generates them on demand in the header during send based on existing parameters (parentID). I think what i suggest actually matches API's behaviour more closely.

Comment (rlejeune74) [reply server/backend/message.go]: Done

**Review by jameshoulahan — APPROVED** [review]

**Pull request merged**